### ISSUE TITLE
Better visualization with respect to hparams

### ISSUE THREAD

**Issue by atishagarwala** [tensorflow/tensorboard#4363]
When doing hyperparameter tuning, i'll often have tensorboards that look like this:

![image](https://user-images.githubusercontent.com/23422890/99750525-68b33200-2a95-11eb-898e-063988ddaca8.png)

which technically has all the information from my tuning run. However, it's hard to parse and use. Ideally, I would be able to:
* Label curves by hparam settings (where I can define which hparams to display).
* Let different hparam choices set the line formatting.

As an example for the latter point: suppose I had a hyperparameter sweep which included different learning rates and different batch sizes. I would love to be able to understand the effect of both by, say:
* Coloring curves with the same learning rates the same color.
* Choosing a different linestyle (solid, dotted, dashed, etc.) for each batch size.
There may be different ways to provide multiple dimensions of curve sorting, IANAE. However I believe having more visualization options can help make tensorboards more useful.

**Comment by psybuzz** [comment]
Thanks for the report!  These are great use cases for us to collect.  Isolating the subset of curves that used a specific hparam value (or range) seems definitely valuable.

**Comment by ssgosh** [comment]
@atishagarwala As a workaround, I perform independent runs of my training scripts using a meta-script. I name training directories with the relevant hparams. Then, I open tensorboard with logdir set to the top-level directory. It is then easy to filter the runs with hparams. However, advanced filtering cannot be done (e.g. some param lambda > 0.1).
